pkg/controllers/user/networkpolicy/policy.go

```diff
@@ -0,0 +1,346 @@
+	networkingv1 "k8s.io/api/networking/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"strings"
+)
+
+const (
+	projectIDAnnotation   = "field.cattle.io/projectId"
+	projectIDNoColonLabel = "management.cattle.io/projectIdNoColon"
+)
+
+type mgr struct {
+	cluster   *config.UserContext
+	pnpLister v3.ProjectNetworkPolicyLister
+	nsLister  typescorev1.NamespaceLister
+}
+
+type projectNetworkPolicyLifecycle struct {
+	mgr *mgr
+}
+
+type projectSyncer struct {
+	mgr *mgr
+}
+
+type projectLifecycle struct {
+	mgr *mgr
+}
+
+type namespaceLifecycle struct {
+	mgr *mgr
+}
+
+func Register(cluster *config.UserContext) {
+	logrus.Infof("Registering project network policy")
+
+	nsLister := cluster.Management.Core.Namespaces("").Controller().Lister()
+	pnpLister := cluster.Management.Management.ProjectNetworkPolicies("").Controller().Lister()
+
+	mgr := &mgr{
+		cluster,
+		pnpLister,
+		nsLister,
+	}
+
+	//plc := &projectLifecycle{mgr: mgr}
+	ps := &projectSyncer{mgr: mgr}
+	nslc := &namespaceLifecycle{mgr: mgr}
+	pnplc := &projectNetworkPolicyLifecycle{mgr: mgr}
+
+	cluster.Management.Management.Projects("").Controller().AddHandler("projectSyncer", ps.Sync)
+	//cluster.Management.Management.Projects("").AddLifecycle("projectLifecycle", plc)
+	cluster.Management.Management.ProjectNetworkPolicies("").AddLifecycle("projectNetworkPolicyLifecycle", pnplc)
+	cluster.Core.Namespaces("").AddLifecycle("namespaceLifecycle", nslc)
+}
+
+func (pnplc *projectNetworkPolicyLifecycle) Create(pnp *v3.ProjectNetworkPolicy) (*v3.ProjectNetworkPolicy, error) {
+	logrus.Infof("pnplc Create pnp=%+v", *pnp)
+
+	projectIDNoColon, ok := pnp.Labels[projectIDNoColonLabel]
+	if !ok {
+		return pnp, fmt.Errorf("couldn't find projectID for pnp=%+v", *pnp)
+	}
+	logrus.Debugf("pnplc Create projectIDNoColon: %v", projectIDNoColon)
+
+	err := programNetworkPolicy(pnplc.mgr, projectIDNoColon)
+	return pnp, err
+}
+
+func (pnplc *projectNetworkPolicyLifecycle) Updated(pnp *v3.ProjectNetworkPolicy) (*v3.ProjectNetworkPolicy, error) {
+	logrus.Infof("pnplc Updated pnp=%+v", pnp)
+
+	projectIDNoColon, ok := pnp.Labels[projectIDNoColonLabel]
+	if !ok {
+		return pnp, fmt.Errorf("couldn't find projectIDNoColon for pnp=%+v", *pnp)
+	}
+	logrus.Debugf("pnplc Updated projectIDNoColon: %v", projectIDNoColon)
+
+	err := programNetworkPolicy(pnplc.mgr, projectIDNoColon)
+	return pnp, err
+}
+
+func (pnplc *projectNetworkPolicyLifecycle) Remove(pnp *v3.ProjectNetworkPolicy) (*v3.ProjectNetworkPolicy, error) {
+	logrus.Infof("pnplc Remove pnp=%+v", pnp)
+
+	projectIDNoColon, ok := pnp.Labels[projectIDNoColonLabel]
+	if !ok {
+		return pnp, fmt.Errorf("couldn't find projectIDNoLabel for pnp=%+v", *pnp)
+	}
+	logrus.Debugf("pnplc Updated projectIDNoColon: %v", projectIDNoColon)
+
+	set := labels.Set(map[string]string{projectIDNoColonLabel: projectIDNoColon})
+
+	err := pnplc.mgr.cluster.K8sClient.NetworkingV1().NetworkPolicies("").DeleteCollection(nil, v1.ListOptions{LabelSelector: set.String()})
```

Comment:
You should be a good citizen and only delete what you create.  Just label your policies.